Updated Makefiles for full MSYS2 and Cygwin installation and testing …

QBos07 opened the pull request:
…support.

They are Linux like environments under Windows and have all the tools needed to support staged installation and testing.

Beware: this only affects the make build system.

Cyan4973 commented on tests/Makefile:
```diff
@@ -287,11 +287,15 @@ ifneq (,$(filter MINGW% MSYS%,$(UNAME)))
 endif
 
+ifneq (,$(filter CYGWIN%,$(UNAME)))
+  HOST_OS = CYGWIN
```
minor: isn't cygwin a `posix` environment ?

QBos07 replied:
Kinda, but in this scenario it should be.

Commit:
Updated Makefiles for full MSYS2 and Cygwin installation and testing support.

They are Linux-like environments under Windows and have all the tools needed to support staged installation and testing.

Beware: this only affects the make build system.
